src/libstore/path.hh

```diff
@@ -73,6 +73,11 @@ const size_t storePathHashLen = 32; // i.e. 160 bits
 /* Extension of derivations in the Nix store. */
 const std::string drvExtension = ".drv";
 
+enum struct FileIngestionMethod : bool {
```

Comment:
Maybe `FileIngestionMethod` should `PathIngestionMethod` since it's more than just files. Or maybe `PathCopyMode` to be a bit less wordy.

Reply:
I wanted something shorter too, but naming it's hard:

 - `PathIngestionMethod` but we are ingesting the file contents, not just the paths

 - `PathCopyMode` sometimes there are no nars involved and we aren't copying anything.

Reply:
One thing that will help is I hope to make a `FileHash` struct for the pair of the ingestion method and the hash; that is a shorter name, and will make a lot of functions take 1 arg less.

Reply:
@edolstra Do you have a name in mind? If not, we've now addressed your other points so I think this ready to go?